WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,231 @@
+				),
+				'siteurl' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'get_bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'add_settings_field' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'add_settings_section' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'register_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'unregister_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'do_action_deprecated' => array(
+			1 => array(
+				'refresh_blog_details' => array(
+					'alt'     => 'the action "clean_site_cache"',
+					'version' => '4.9.0',
+				),
+			),
+		),
+		'do_action_deprecated' => array(
+			1 => array(
+				'blog_details' => array(
+					'alt'     => 'the filter "site_details"',
+					'version' => '4.7.0',
+				),
+				'rest_enabled' => array(
+					'alt'     => 'the rest_authentication_errors filter to restrict access to the REST API',
+					'version' => '4.7.0',
+				),
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$param_count = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $paramerter_args ) {
+
+			if ( $position > $param_count ) {
+				break;
+			}
+
+			if ( ! isset( $parameters[ $position ] ) ) {
```

Comment:
Can you give me a code example explaining what this is testing ?

AFAICS, this could only ever happen if a) the `$position > $param_count` as is already tested above or b) when no parameter is passed for one of the preceding parameters which would always cause a parse error.